src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -944,10 +945,10 @@ private void AddDefaultValue(EntityHandle parentHandle, object? defaultValue)
                 bodyOffset: offset,
                 parameterList: MetadataTokens.ParameterHandle(parameterToken));
 
-        private TypeReferenceHandle AddTypeReference(Type type, EntityHandle resolutionScope) =>
+        private TypeReferenceHandle AddTypeReference(Type type, EntityHandle resolutionScope, string? ns) =>
```

Comment:
```suggestion
        private TypeReferenceHandle AddTypeReference(EntityHandle resolutionScope, string name, string? ns) =>
```
The method should either take the namespace/name strings or the Type, but not mix-match.